scanomatic/models/validators/validate.py

```diff
@@ -0,0 +1,92 @@
+from collections.abc import Iterator
+from enum import Enum, auto
+from typing import Literal, Optional, Type, Union, cast
+
+from scanomatic.generics.model import Model
+from scanomatic.models.factories.factory_lookup import get_factory
+from scanomatic.models.fields_lookup import get_field_types
+
+from .validation_lookup import get_special_validators
+
+ValidationResults = Iterator[Union[Literal[True], Enum]]
+
+
+class UnknownFields(Enum):
+    UNKNOWN_FIELD = auto()
+
+
+def _produce_validation_error(fields: Optional[Type[Enum]], field: str) -> Enum:
+    if fields is None:
+        return UnknownFields.UNKNOWN_FIELD
+    if field == "name":
+        return fields["_name"]
```

Comment:
What is the logic here?

Reply:
enums already have `name` attribute so it's really problematic that there were enum values called `name` too... this works around that.